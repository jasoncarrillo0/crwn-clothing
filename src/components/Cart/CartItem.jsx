import React from 'react';
import styles from './CartItem.module.scss';

function CartItem( { item: { imageUrl, price, name, quantity } }) {
    return (
        <div className={styles.cartItem}>
            <img src={imageUrl} alt="item"/>
            <div className={styles.itemDetails}>
                <span className={styles.name}>{name}</span>
                <span className="price">{quantity} x ${price}</span>
            </div>
        </div>
    )
}
export default CartItem;