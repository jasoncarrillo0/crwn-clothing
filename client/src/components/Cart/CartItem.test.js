import { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import CartItem from './CartItem';
configure({adapter: new Adapter()});

let wrapper;
beforeEach(() => {
    const mockProps = {
        item: {
            imageUrl: 'https://images-na.ssl-images-amazon.com/images/I/51XX3NKAVAL._AC_.jpg',
            price: 20,
            name: "Blue hat",
            quantity: 2
        }
    }
    wrapper = shallow(<CartItem {...mockProps}/>)
})

it('expect to render CartItem', () => {
    expect(wrapper).toMatchSnapshot();
});
