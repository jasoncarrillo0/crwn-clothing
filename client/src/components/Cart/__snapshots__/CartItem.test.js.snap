// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CartItem 1`] = `ShallowWrapper {}`;
